### PR TITLE
Tests: replace express with basic Node server (3.x)

### PR DESCRIPTION
**Pull request by timmywil**
### Summary ###

3.x version of #5527 and #5532 

- replace express with basic Node server
- Replace dead links in qunit fixture and use https for all links.
- Upgrade dependencies (not sinon or uglify)

Sample browserstack run: https://github.com/timmywil/jquery/actions/runs/10353380768

### Checklist ###

* [x] New tests have been added to show the fix or feature works
* [x] If needed, a docs issue/PR was created at https://github.com/jquery/api.jquery.com

### CODE REVIEW

File: eslint.config.js [hunk 1]
```diff
@@ -271,7 +279,7 @@ module.exports = [
 			"test/node_smoke_tests/**",
 			"test/bundler_smoke_tests/**/*",
 			"test/promises_aplus_adapters/**",
-			"test/middleware-mockserver.js"
+			"test/middleware-mockserver.cjs"
```
Comment (timmywil) [inline eslint.config.js:282]: I think we had said it'd be nice if this filename matched the main branch, so I've made that change in this PR.

File: package.json [hunk 2]
```diff
@@ -61,35 +61,33 @@
   "license": "MIT",
   "devDependencies": {
     "@babel/cli": "7.24.8",
-    "@babel/core": "7.24.9",
+    "@babel/core": "7.25.2",
     "@babel/plugin-transform-for-of": "7.24.7",
     "@prantlf/jsonlint": "14.0.3",
     "@types/selenium-webdriver": "4.1.24",
-    "body-parser": "1.20.2",
+    "archiver": "7.0.1",
```
Comment (timmywil) [inline package.json:68]: This was missing from the release PR.

File: test/data/mock.php [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <?php
 
 /**
- * Keep in sync with /test/middleware-mockserver.js
+ * Keep in sync with /test/middleware-mockserver.cjs
```
Comment (mgol) [inline test/data/mock.php:4]: BTW, `mock.php` on `main` has some differences from the `3.x-stable` version, mostly due to response formatting (3.x has more spaces). Spacey versions better suite our code style, but we should align regardless. `middleware-mockserver.cjs` has way fewer changes - mostly extra stuff for `main` which makes sense.

File: test/runner/createTestServer.js [hunk 4]
```diff
@@ -1,70 +1,192 @@
...
+			req.method !== "HEAD" &&
+			( ext !== ".html" || req.method !== "POST" )
+		) {
+			return next();
+		}
+		const mimeType = validMimeTypes[ ext ];
+		if ( mimeType ) {
+			try {
+				await stat( file );
+			} catch ( _ ) {
```
Comment (mgol) [inline test/runner/createTestServer.js:171]: Are we going to align `main` here?

**Review by mgol — APPROVED** [review]
LGTM. Just a few questions about alignment, but not strictly related to this PR landing.

File: test/data/mock.php (hunk 3 above)
Comment (timmywil) [reply test/data/mock.php:4]: I've copied what changes I thought made sense from main, but not the spacing. We can then switch the spacing on main.

File: test/data/mock.php [hunk 5]
```diff
@@ -196,22 +215,24 @@ protected function testHTML( $req ) {
...
 		header( 'Content-type: text/html' );
 		echo file_get_contents( __DIR__ . '/csp-nonce' . $test . '.html' );
 	}
 
+	protected function cspAjaxScript( $req ) {
+		header( "Content-Security-Policy: script-src 'self'; report-uri ./mock.php?action=cspLog" );
+		header( 'Content-type: text/html' );
+		echo file_get_contents( __DIR__ . '/csp-ajax-script.html' );
+	}
+
```
Comment (mgol) [inline test/data/mock.php:235]: This doesn't make sense here; the referenced file doesn't even exist on `3.x-stable`.

**Review by mgol — CHANGES_REQUESTED** [review]

File: test/data/mock.php (hunk 5 above)
Comment (timmywil) [reply test/data/mock.php]: Removed

**Review by mgol — APPROVED** [review]

**Pull request merged**